variation-commons-mongodb/src/test/java/uk/ac/ebi/eva/commons/mongodb/configuration/MongoRepositoryTestConfiguration.java

```diff
@@ -18,85 +18,76 @@
  */
 package uk.ac.ebi.eva.commons.mongodb.configuration;
 
-import com.github.fakemongo.Fongo;
-import com.mongodb.MongoClient;
 import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.boot.test.autoconfigure.data.mongo.AutoConfigureDataMongo;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.context.annotation.Import;
-import org.springframework.context.annotation.PropertySource;
-import org.springframework.data.mongodb.MongoDbFactory;
-import org.springframework.data.mongodb.core.MongoTemplate;
-import org.springframework.data.mongodb.core.SimpleMongoDbFactory;
-import org.springframework.data.mongodb.core.convert.DefaultMongoTypeMapper;
-import org.springframework.data.mongodb.core.convert.MappingMongoConverter;
+import org.springframework.data.mongodb.config.EnableMongoAuditing;
+import org.springframework.data.mongodb.repository.config.EnableMongoRepositories;
 import org.springframework.util.Assert;
 
 @Configuration
-@Import({EvaRepositoriesConfiguration.class})
-@PropertySource({"classpath:eva.properties"})
+@EntityScan(basePackages = {"uk.ac.ebi.eva.commons.mongodb.entities"})
+@EnableMongoRepositories(basePackages = "uk.ac.ebi.eva.commons.mongodb.repositories")
+@EnableMongoAuditing
+@AutoConfigureDataMongo
 public class MongoRepositoryTestConfiguration {
 
+    @Value("${mongodb.read-preference}")
+    private String readPreference;
+
+    @Value("${eva.mongo.collections.annotation-metadata:#{null}}")
+    private String collectionAnnotationMetadata;
+
+    @Value("${eva.mongo.collections.annotations:#{null}}")
+    private String collectionAnnotations;
+
+    @Value("${eva.mongo.collections.features:#{null}}")
+    private String collectionFeatures;
+
+    @Value("${eva.mongo.collections.variants:#{null}}")
+    private String collectionVariants;
+
+    @Value("${eva.mongo.collections.files:#{null}}")
+    private String collectionFiles;
+
+    @Value("${eva.mongo.collections.samples:#{null}}")
+    private String collectionSamples;
+
     @Bean
-    public String mongoCollectionsAnnotationMetadata(
-            @Value("${eva.mongo.collections.annotation-metadata:#{null}}") String collectionAnnotationMetadata) {
```

Comment:
did this not work? or was it a style decision to move the value injection to class fields?

Reply:
I was experimenting with many things when there were autowiring error messages that were driving me crazy :D. Looks like the older style still works. Will restore these to the older style...